src/command/init/mod.rs

```diff
@@ -76,39 +80,86 @@ impl Init {
 
         let project_type_selected = welcome.select_project_type(&self.project_type, &self.path)?;
 
-        match project_type_selected.project_type {
-            ProjectType::CreateNew => {
-                let use_case_selected_option = project_type_selected
-                    .select_organization(&self.organization, &self.profile, &client_config)
+        // Early return for AddSubgraph case
+        if project_type_selected.project_type == ProjectType::AddSubgraph {
+            display_use_template_message();
+            return Ok(RoverOutput::EmptySuccess);
+        }
+
+        // Handle new project creation flow
+        let use_case_selected = match project_type_selected
+            .select_organization(&self.organization, &self.profile, &client_config)
+            .await?
+            .select_use_case(&self.project_use_case)?
+        {
+            Some(use_case) => use_case,
+            None => return Ok(RoverOutput::EmptySuccess),
+        };
+
+        let creation_confirmed = match use_case_selected
+            .enter_project_name(&self.project_name)?
+            .confirm_graph_id(&self.graph_id)?
+            .preview_and_confirm_creation(http_service.clone())
+            .await?
+        {
+            Some(confirmed) => confirmed,
+            None => return Ok(RoverOutput::EmptySuccess),
+        };
+
+        let project_created = creation_confirmed
+            .create_project(&client_config, &self.profile)
+            .await?;
+
+        // Handle project creation result
+        if let CreateProjectResult::Created(project) = project_created {
+            return Ok(project.complete().success());
+        }
+
+        // Handle restart loop
+        if let CreateProjectResult::Restart(mut current_project) = project_created {
+            const MAX_RETRIES: u32 = 3;
+            let mut retry_count = 0;
+
+            loop {
```

Comment:
nit: you can use this pattern here and won't need the retry_count mutable


```suggestion
            for attempt in 0..=MAX_RETRIES {
```

Reply:
YES. I did this for another loop (that I deprecated) and you're right, this pattern is great 👍